src/Controller/ApiController.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\UnsupportedMediaTypeHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Routing\RouterInterface;
+
+class ApiController
+{
+    private array $products;
+
+    /**
+     * @param Request         $request
+     * @param RouterInterface $router
+     *
+     * @return Response
+     */
+    #[Route('/api', name: 'resource_list', ),]
+    public function getResourceList(Request $request, RouterInterface $router): Response
+    {
+        if (!$request->headers->get('content-type')) {
+            throw new UnsupportedMediaTypeHttpException('Your request must contain a Content-type header.');
+        }
+
+        $resources = [];
+        $routes = $router->getRouteCollection()->all();
```

Comment:
For performance reasons, it's better to not use the route collection but collect these routes manually and use the generator interface instead.